public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsEdit.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { Checkbox, CollapsableSection, ColorValueEditor, Field, HorizontalGroup, Input } from '@grafana/ui';
+import { DashboardModel } from '../../state/DashboardModel';
+import { AnnotationQuery, DataSourceInstanceSettings } from '@grafana/data';
+import { DataSourcePicker } from 'app/core/components/Select/DataSourcePicker';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { useAsync } from 'react-use';
+import StandardAnnotationQueryEditor from 'app/features/annotations/components/StandardAnnotationQueryEditor';
+import { AngularEditorLoader } from './AngularEditorLoader';
+
+export const newAnnotation: AnnotationQuery = {
+  name: 'New annotation',
+  enable: true,
+  datasource: null,
+  iconColor: 'red',
+};
+
+type Props = {
+  editIdx: number | null;
+  dashboard: DashboardModel;
+};
+
+export const AnnotationSettingsEdit: React.FC<Props> = ({ editIdx, dashboard }) => {
+  const [annotation, setAnnotation] = useState(editIdx !== null ? dashboard.annotations.list[editIdx] : newAnnotation);
+
+  const { value: ds } = useAsync(async () => {
+    return await getDataSourceSrv().get(annotation.datasource);
+  }, [annotation.datasource]);
+
+  const onUpdate = (annotation: AnnotationQuery) => {
+    const list = [...dashboard.annotations.list];
+    list.splice(editIdx!, 1, annotation);
+    setAnnotation(annotation);
+    dashboard.annotations.list = list;
+  };
+
+  const onNameChange = (ev: React.FocusEvent<HTMLInputElement>) => {
+    onUpdate({
+      ...annotation,
+      name: ev.currentTarget.value,
+    });
+  };
+
+  const onDataSourceChange = (ds: DataSourceInstanceSettings) => {
+    onUpdate({
+      ...annotation,
+      datasource: ds.name,
+    });
+  };
+
+  const onChange = (ev: React.FocusEvent<HTMLInputElement>) => {
+    const target = ev.currentTarget;
+    onUpdate({
+      ...annotation,
+      [target.name]: target.type === 'checkbox' ? target.checked : target.value,
+    });
+  };
+
+  const onColorChange = (color: string) => {
+    onUpdate({
+      ...annotation,
+      iconColor: color,
+    });
+  };
+
+  const isNewAnnotation = annotation.name === newAnnotation.name;
+
+  return (
+    <div>
+      <Field label="Name">
+        <Input
+          name="name"
```

Comment:
```suggestion
          name="name"
          id="name"
```
Good idea to add `id` so the label is properly associated. It also helps later in the tests as it allows matching by role with a `name` parameter.